pinot-core/src/test/java/com/linkedin/pinot/core/query/scheduler/MultiLevelPriorityQueueTest.java

```diff
@@ -0,0 +1,235 @@
+/**
+ * Copyright (C) 2014-2016 LinkedIn Corp. (pinot-core@linkedin.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.linkedin.pinot.core.query.scheduler;
+
+import com.google.common.base.Preconditions;
+import com.linkedin.pinot.common.metrics.ServerMetrics;
+import com.linkedin.pinot.core.query.scheduler.resources.PolicyBasedResourceManager;
+import com.linkedin.pinot.core.query.scheduler.resources.ResourceLimitPolicy;
+import com.linkedin.pinot.core.query.scheduler.resources.ResourceManager;
+import com.linkedin.pinot.core.query.scheduler.resources.UnboundedResourceManager;
+import com.yammer.metrics.core.MetricsRegistry;
+import java.util.concurrent.BrokenBarrierException;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.concurrent.CyclicBarrier;
+import org.apache.commons.configuration.Configuration;
+import org.apache.commons.configuration.PropertiesConfiguration;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import static com.linkedin.pinot.core.query.scheduler.TestHelper.*;
+import static org.testng.Assert.*;
+
+
+public class MultiLevelPriorityQueueTest {
+  SchedulerGroup group;
+  final ServerMetrics metrics = new ServerMetrics(new MetricsRegistry());
+
+  final SchedulerGroupMapper groupMapper = new TableBasedGroupMapper();
+  final TestSchedulerGroupFactory groupFactory = new TestSchedulerGroupFactory();
+  final String groupOne = "1";
+  final String groupTwo = "2";
+
+  @BeforeMethod
+  public void beforeMethod() {
+    groupFactory.reset();
+  }
+
+  @Test (expectedExceptions = IllegalArgumentException.class)
+  public void testPutNull() throws OutOfCapacityError {
+    createQueue().put(null);
+  }
+
+  @Test
+  public void testSimplePutTake() throws OutOfCapacityError {
+    MultiLevelPriorityQueue queue = createQueue();
+
+    queue.put(createQueryRequest(groupOne, metrics));
+    queue.put(createQueryRequest(groupTwo, metrics));
+    queue.put(createQueryRequest(groupOne, metrics));
+
+    assertEquals(groupFactory.numCalls.get(), 2);
+    SchedulerQueryContext r = queue.take();
+    assertEquals(r.getSchedulerGroup().name(), groupOne);
+    r = queue.take();
+    assertEquals(r.getSchedulerGroup().name(), groupOne);
+    r = queue.take();
+    assertEquals(r.getSchedulerGroup().name(), groupTwo);
+  }
+
+  @Test (expectedExceptions = OutOfCapacityError.class)
```

Comment:
i thought we agreed to remove this in favor of testing for the exception at the specific line we expect to get it?